components/org.wso2.carbon.identity.oidc.session/src/main/java/org/wso2/carbon/identity/oidc/session/OIDCSessionManager.java

```diff
@@ -32,69 +34,167 @@
 
+        String tenantDomain = resolveCacheTenantDomain(loginTenantDomain);
         OIDCSessionParticipantCacheKey cacheKey = new OIDCSessionParticipantCacheKey();
         cacheKey.setSessionID(sessionId);
 
         OIDCSessionParticipantCacheEntry cacheEntry = OIDCSessionParticipantCache.getInstance().getValueFromCache
-                (cacheKey);
+                (cacheKey, tenantDomain);
 
         return cacheEntry == null ? null : cacheEntry.getSessionState();
     }
 
     /**
      * Removes the session against the old session id and restore against the provided new session id.
      *
-     * @param oldSessionId
-     * @param newSessionId
-     * @param sessionState
+     * @param oldSessionId old session id value
+     * @param newSessionId new session id value
+     * @param sessionState OIDCSessionState instance
+     * @param loginTenantDomain login tenant domain
      */
-    public void restoreOIDCSessionState(String oldSessionId, String newSessionId, OIDCSessionState sessionState) {
+    public void restoreOIDCSessionState(String oldSessionId, String newSessionId, OIDCSessionState sessionState,
+                                        String loginTenantDomain) {
 
-        removeOIDCSessionState(oldSessionId);
-        storeOIDCSessionState(newSessionId, sessionState);
+        String tenantDomain = resolveCacheTenantDomain(loginTenantDomain);
+        removeOIDCSessionState(oldSessionId, tenantDomain);
+        storeOIDCSessionState(newSessionId, sessionState, tenantDomain);
     }
 
     /**
      * Removes the session against the given session id.
      *
      * @param sessionId session id value
+     * @param loginTenantDomain login tenant domain
      */
-    public void removeOIDCSessionState(String sessionId) {
+    public void removeOIDCSessionState(String sessionId, String loginTenantDomain) {
 
+        String tenantDomain = resolveCacheTenantDomain(loginTenantDomain);
         OIDCSessionParticipantCacheKey cacheKey = new OIDCSessionParticipantCacheKey();
         cacheKey.setSessionID(sessionId);
 
-        OIDCSessionParticipantCache.getInstance().clearCacheEntry(cacheKey);
+        OIDCSessionParticipantCache.getInstance().clearCacheEntry(cacheKey, tenantDomain);
     }
 
     /**
      * Checks if there is a session exists for the gives session id.
      *
      * @param sessionId session id value
+     * @param loginTenantDomain login tenant domain
      * @return true if session exists
      */
-    public boolean sessionExists(String sessionId) {
+    public boolean sessionExists(String sessionId, String loginTenantDomain) {
+
+        String tenantDomain = resolveCacheTenantDomain(loginTenantDomain);
+        return getOIDCSessionState(sessionId, tenantDomain) != null;
+    }
+
+    private String resolveCacheTenantDomain(String tenantDomain) {
 
-        return getOIDCSessionState(sessionId) != null;
+        if (!IdentityTenantUtil.isTenantedSessionsEnabled()) {
```

Comment:
Its good to have explained if-else clause but here can't we use one-line logic using ? and :

Reply:
For the readability purposes i think it is better to have this explained if-else clause